src/Pages/Explorer.elm

```diff
@@ -1,54 +1,82 @@
 module Pages.Explorer exposing (Model, Msg, init, page, update, view)
 
-import Compute
+import Browser.Navigation as Navigation
+import Compute exposing (Msg(..), Tab(..))
+import Dict
 import Element as E
 import Element.Font as Font
 import Element.Region as Region
 import Layout
 import Page
 import Request exposing (Request)
 import Shared
+import Url
+import Url.Builder
 import Url.Parser exposing (..)
 import View exposing (View)
 
 
 type alias Model =
-    { wordCloud : Compute.Model
-    }
+    Compute.Model
 
 
 type Msg
-    = WordCloudMsg Compute.Msg
+    = Explorer Compute.Msg
+
+
+type alias UrlParams =
+    -- Values in the URL that may initialize query or tab
+    { query : Maybe String
+    , tab : Maybe Tab
+    }
 
 
 page : Shared.Model -> Request -> Page.With Model Msg
-page shared _ =
+page shared ({ query } as req) =
     Page.element
-        { init = init shared
-        , update = update
+        { init = init shared (decodeUrlParams query)
+        , update = update req
         , view = view
         , subscriptions = subscriptions
         }
 
 
-init : Shared.Model -> ( Model, Cmd Msg )
-init shared =
+init : Shared.Model -> UrlParams -> ( Model, Cmd Msg )
+init shared urlParams =
+    Compute.init shared
+        |> Tuple.mapBoth
+            (\model ->
+                { model
+                    | query = Maybe.withDefault model.query urlParams.query
+                    , selectedTab = Maybe.withDefault model.selectedTab urlParams.tab
+                }
+            )
+            (Cmd.map Explorer)
+
+
+update : Request -> Msg -> Model -> ( Model, Cmd Msg )
+update req msg model =
     let
-        ( subModel, subCmd ) =
-            Compute.init shared
-    in
-    ( { wordCloud = subModel }, Cmd.map WordCloudMsg subCmd )
+        ( computeModel, computeCmd ) =
+            case msg of
+                Explorer subMsg ->
+                    Compute.update subMsg model
 
+        explorerCmd =
+            -- Explorer page doesn't modify the model, just possibly a command (triggers URL update)
+            case msg of
+                Explorer subMsg ->
+                    case subMsg of
+                        OnTabSelected tab ->
+                            Navigation.replaceUrl req.key (encodeNewUrl model.query tab)
```

Comment:
I'm using `Navigation` to change the URL, I couldn't figure out how to do it easily with elm-spa routing. There's probably a way (and elm-spa probably uses `Navigation` underneath the hood anyway) but this plays well enough